docs/doc.md

```diff
@@ -0,0 +1,186 @@
+# Application
+
+This doc explains high-level app design with a structural and functional view. 
+
+## Structural View
+
+```mermaid
+---
+title: Layer View
+---
+graph
+    UI --> Data
+    UI --> Controllers --> Data
+    x[Cross-Layer]
+```
+
+`UI` implements everything related to interaction with the user: screens, animations, navigation between the screens.
+
+`Controllers` implement processes and business logic functionality of the app, uses `Data` layer's objects.
+
+`Data` implements access to the persistence, secure storage, as well as API models for various remote services. 
+
+### UI
+
+```mermaid
+---
+title: UI
+---
+graph
+    SceneDelegate --> Window --> ViewController
+    Flow --> ViewController --> Data
+    ViewController --> View
+    Flow --> Data
+```
+
+`Flow` implements navigation logic between different screens. It connects `ViewController`s together and passes data around them. 
+
+`ViewController` implements single screen logic using `View`s as components and data provided from outside (usually a `Flow` or another `ViewController`). 
+
+Both `Flow`s and `ViewController`s can use `Data` layer to implement conditional logic in the interface.
+
+### Data
+```mermaid
+---
+title: Data
+---
+graph
+    Data --> Models[Logic Models]
+    Data --> CD[Core Data Models]
+    Data --> Services
+    Services --> CGW[Client Gateway] --> APIModels[API Models]
+    Services --> NodeRPC
+    Services --> Relay
+    Services --> WalletConnect
+    Services --> Moonpay    
+    Services --> Web3Auth
```

Comment:
Does this arrow represent a dependency or that it "contains" the respective components?